scripts/core/editor3/directive.tsx

```diff
@@ -519,6 +529,20 @@ class Editor3Directive {
                     renderEditor3();
                 });
 
+                $scope.$on('formattingOptions-update', (_e, data) => {
```

Comment:
what about adding a watcher instead that would look at `editorFormat` changes? Then it would apply universally - also for authoring which is not the case now. It would also be safer to dispatch an action vs reinitialize the entire store.